frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,256 @@
+          <div>
+            This step corresponds to execution{' '}
+            <Link
+              className={commonCss.link}
+              to={RoutePageFactory.executionDetails(execution.getId())}
+            >
+              "{ExecutionHelpers.getName(execution)}".
+            </Link>
+          </div>
+          {!enableFetch && <Banner message='Node has not completed.' mode='info' />}
+          {enableFetch && isLoadingArtifactTypes && isLoadingArtifacts && (
+            <Banner message='Metrics is loading.' mode='info' />
+          )}
+          {enableFetch && errorArtifacts && (
+            <Banner
+              message='Error in retrieving metrics information.'
+              mode='error'
+              additionalInfo={errorArtifacts.message}
+              refresh={refresh}
+            />
+          )}
+          {enableFetch && !errorArtifacts && errorArtifactTypes && (
+            <Banner
+              message='Error in retrieving artifact types information.'
+              mode='error'
+              additionalInfo={errorArtifactTypes.message}
+              refresh={refresh}
+            />
+          )}
+          {enableFetch &&
+            isSuccessArtifacts &&
+            isSuccessArtifactTypes &&
+            artifacts &&
+            artifactTypes && (
+              <MetricsSwitcher
+                artifacts={artifacts}
+                artifactTypes={artifactTypes}
+                refresh={refresh}
+              ></MetricsSwitcher>
+            )}
+        </div>
+      </div>
+    </ErrorBoundary>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
+
+  const confidenceMetricsArtifacts = classificationMetricsArtifacts
+    .map(artifact => ({
+      id: artifact.getId(),
+      name: artifact
+        .getCustomPropertiesMap()
+        .get('name')
+        ?.getStringValue(),
+      customProperties: artifact.getCustomPropertiesMap(),
+    }))
+    .filter(x => !!x.name)
+    .filter(x => {
+      const confidenceMetrics = x.customProperties
+        .get('confidenceMetrics')
+        ?.getStructValue()
+        ?.toJavaScript();
+      return !!confidenceMetrics;
+    });
+
+  return (
+    <>
+      {(!confidenceMetricsArtifacts || confidenceMetricsArtifacts.length === 0) && (
+        <Banner message='There is no metrics artifact available in this step.' mode='info' />
+      )}
+      {confidenceMetricsArtifacts &&
+        confidenceMetricsArtifacts.length > 0 &&
+        confidenceMetricsArtifacts.map(artifact => {
```

Comment:
I agree that wrapping to react component is helpful. I am thinking about how does the API look like for this new component: For example, we can give an option to choose what metrics to show, and what kind of props to provide. It will make more sense when we have multiple metrics to show for one artifact. I will do it in next PR.

Reply:
SGTM